test/scripts/e2e_client_runner.py

```diff
@@ -271,14 +277,16 @@ def wait(self, timeout):
 already_stopped = False
 already_deleted = False
 
-def goal_network_stop(netdir, normal_cleanup=False):
+def goal_network_stop(netdir, env, normal_cleanup=False):
     global already_stopped, already_deleted
     if already_stopped or already_deleted:
         return
 
     logger.info('stop network in %s', netdir)
     try:
         xrun(['goal', 'network', 'stop', '-r', netdir], timeout=10)
+        algodata = env['ALGORAND_DATA']
+        xrun(['goal', 'kmd', 'stop', '-d', algodata], timeout=5)
```

Comment:
move into a separate try/catch block to prevent kmd keep running if network stopping falied.

Reply:
done.